remote_attestation/rust/src/crypto.rs

```diff
@@ -136,24 +137,28 @@ impl AeadEncryptor {
     /// https://briansmith.org/rustdoc/ring/aead/constant.NONCE_LEN.html
     fn generate_nonce() -> [u8; aead::NONCE_LEN] {
         let mut nonce: [u8; aead::NONCE_LEN] = Default::default();
-        let rng = SystemRandom::new();
-        rng.fill(&mut nonce).unwrap();
+        let mut random = get_random(aead::NONCE_LEN);
+        nonce.copy_from_slice(&mut random[0..aead::NONCE_LEN]);
         nonce
     }
 }
 
 /// Implementation of the X25519 Elliptic Curve Diffie-Hellman (ECDH) key negotiation.
 /// https://datatracker.ietf.org/doc/html/rfc7748#section-6.1
 pub struct KeyNegotiator {
+    r#type: KeyNegotiatorType,
     private_key: agreement::EphemeralPrivateKey,
 }
 
 impl KeyNegotiator {
-    pub fn create() -> anyhow::Result<Self> {
+    pub fn create(r#type: KeyNegotiatorType) -> anyhow::Result<Self> {
```

Comment:
I think it is more idiomatic to just use a non-conflicting name for the parameter, e.g. `type_`

Reply:
Done